src/Playwright/Transport/Connection.cs

```diff
@@ -430,5 +420,89 @@ internal void TraceMessage(string logLevel, object message)
                 Console.Error.WriteLine(line);
             }
         }
+
+        internal async Task<T> WrapApiCallAsync<T>(Func<Task<T>> action, bool isInternal = false)
+        {
+            EnsureApiZoneExists();
+            if (ApiZone.Value[0] != null)
+            {
+                return await action().ConfigureAwait(false);
+            }
+            var st = new StackTrace(true);
+            var stack = new List<object>();
+            var lastInternalApiName = string.Empty;
+            var apiName = string.Empty;
+            for (int i = 0; i < st.FrameCount; ++i)
+            {
+                var sf = st.GetFrame(i);
+                string fileName = sf.GetFileName();
+                string cSharpNamespace = sf.GetMethod().ReflectedType.Namespace;
+                bool playwrightInternal = cSharpNamespace == "Microsoft.Playwright" || cSharpNamespace.StartsWith("Microsoft.Playwright.Core", StringComparison.InvariantCultureIgnoreCase) || cSharpNamespace.StartsWith("Microsoft.Playwright.Transport", StringComparison.InvariantCultureIgnoreCase) || cSharpNamespace.StartsWith("Microsoft.Playwright.Helpers", StringComparison.InvariantCultureIgnoreCase);
```

Comment:
I'm not a big fan of it, but we can't go via the fileName, since its not always there, only ReflectedType works then.
We need to ensure that its not our nunit and mstest harness and not our tests which have similar namespaces.